cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc

```diff
@@ -806,6 +833,32 @@ Please instead use:
 ----
 point.withinBBox(point({x:1, y:1}), point({x:0, y:0}), point({x:2, y:2}))
 ----
+
+a|
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+MATCH (n)
+RETURN
+CASE n.prop
+    WHEN null THEN 'one'
+    ELSE 'two'
+END
+----
+Currently, if n.prop is null, 'one' would be returned.
+a|
+The matching of null == null in CASE expressions is deprecated and in future versions will no longer be able to be matched on.
+Please use `IS NULL` instead.
+[source, cypher, role="noheader"]
+----
+MATCH (n)
+RETURN
+CASE
+    WHEN n.prop IS NULL THEN 'one'
+    ELSE 'two'
+END
+----
```

Comment:
```suggestion
a|
label:functionality[]
label:deprecated[]
[source, cypher, role="noheader"]
----
MATCH (n)
RETURN
CASE n.prop
    WHEN null THEN 'one'
    ELSE 'two'
END
----
a|
Currently, if `n.prop` is `null`, `'one'` would be returned.
Since `null = null` returns `false` in Cypher, a `WHEN` expression will no longer match in future versions.

Please use `IS NULL` instead:
[source, cypher, role="noheader"]
----
MATCH (n)
RETURN
CASE
    WHEN n.prop IS NULL THEN 'one'
    ELSE 'two'
END
----
```

Reply:
Replicating the wording in #1555